pkg/client/cache/eventqueue.go

```diff
@@ -324,11 +339,22 @@ func (eq *EventQueue) Replace(objects []interface{}, resourceVersion string) err
 	}
 
 	if len(eq.queue) > 0 {
+		eq.lastReplaceKey = eq.queue[len(eq.queue)-1]
 		eq.cond.Broadcast()
+	} else {
+		eq.lastReplaceKey = ""
 	}
 	return nil
 }
 
+// TODO doc
+func (eq *EventQueue) LastReplaceRead() bool {
```

Comment:
Oh a straight "getter" style name that doesn't get the value?  Yeah, more obvious name please.